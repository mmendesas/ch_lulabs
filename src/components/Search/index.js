import React, { Component } from 'react';

class Search extends Component {
  state = {
    cep: ''
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState(() => ({ [name]: value }));
  };

  submitForm = e => {
    e.preventDefault();
    this.props.onSearch(this.state.cep);
    this.setState(() => ({ cep: '' }));
  };

  render() {
    return (
      <div className="search">
        <h2 className="search__title">Consultar</h2>
        <form onSubmit={this.submitForm}>
          <label className="search__label">CEP</label>
          <input
            type="text"
            placeholder="02050-010"
            name="cep"
            className="search__text"
            value={this.state.cep}
            onChange={this.handleChange}
          />
          <button type="submit" className="search__button">
            Buscar
          </button>
        </form>
      </div>
    );
  }
}

export default Search;
