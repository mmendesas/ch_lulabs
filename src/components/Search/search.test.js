import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import Search from './index';

describe('[Component] - Search', () => {
  const setup = () => {
    const onSearch = jest.fn();
    const utils = render(<Search onSearch={onSearch} />);
    const input = utils.getByLabelText('search');
    const button = utils.container.querySelector('.search__button');

    return {
      input,
      button,
      onSearch,
      ...utils
    };
  };

  it('should be match snapshot', () => {
    const { getByText } = render(<Search />);
    expect(getByText('Consultar')).toMatchSnapshot();
  });

  it('should allow the numeric postal code in the search entry', () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: '02047000' } });
    expect(input.value).toBe('02047000');
  });

  it('should not allow letters to be inputted', () => {
    const { input } = setup();
    expect(input.value).toBe(''); // empty before
    fireEvent.change(input, { target: { value: 'Good Day' } });
    expect(input.value).toBe(''); //empty after
  });

  it('should call onSearch function propertie', () => {
    const { button, onSearch } = setup();
    fireEvent.click(button);
    expect(onSearch).toHaveBeenCalled();
  });
});
