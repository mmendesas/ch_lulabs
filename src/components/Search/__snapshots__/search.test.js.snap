// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - Search should be match snapshot 1`] = `
<h2
  class="search__title"
>
  Consultar
</h2>
`;
